sre-server/main.go

```diff
@@ -42,23 +103,218 @@ func run(args []string) error {
 		return err
 	}
 
+	// Create a shared informer factory
+	// TODO add defaultResync to the configuration
+	sharedInformers := informers.NewSharedInformerFactory(clientset, time.Minute*5)
+
+	// Create a deployment informer for all namespaces
+	deploymentInformer := sharedInformers.Apps().V1().Deployments().Informer()
+
+	// create a deployment lister for all namespaces
+	deploymentLister := sharedInformers.Apps().V1().Deployments().Lister()
+
+	// Start the informer factory to begin watching for changes
+	// we run the informer for the lifetime of the application
+	sharedInformers.Start(context.Background().Done())
+
+	// Wait for the cache to sync before exiting
+	if !cache.WaitForCacheSync(context.Background().Done(), deploymentInformer.HasSynced) {
+		log.Panicln("Timed out waiting for caches to sync")
+	}
+	log.Println("Informer is started")
+	// add router
+	router := mux.NewRouter()
 	// add handlers
-	http.Handle("/healthz", &healthzHandler{clientset: clientset})
+	router.Handle("/api/v1/namespaces/{namespace}/deployments/{name}/replicas",
+		&deploymentHandler{clientset: clientset, lister: deploymentLister})
+	router.Handle("/api/v1/deployments", &deploymentsHandler{lister: deploymentLister})
+	router.Handle("/api/v1/healthz", &healthzHandler{clientset: clientset, informer: deploymentInformer})
+	router.Handle("/api/v1/pingz", &pingzHandler{})
+
+	// Create HTTP Server with TLS Config
+	server := &http.Server{
+		Addr:      ":" + port,
+		TLSConfig: tlsConfig,
+		Handler:   router,
+	}
+	log.Printf("Starting HTTPS server on port %s...", port)
+
+	return server.ListenAndServeTLS("", "") // Certificates are in the TLSConfig
+}
+
+// deploymentHandler is an HTTP handler for the deployment API.
+type deploymentHandler struct {
+	clientset *kubernetes.Clientset
+	lister    listersv1.DeploymentLister
+}
+
+func writeMessage(w http.ResponseWriter, statusCode int, data any) {
+	d, _ := json.Marshal(data)
+	w.WriteHeader(statusCode)
+	_, err := w.Write(d)
+	if err != nil {
+		log.Printf("unable to write response: %v", err)
+	}
+}
+
+func writeError(w http.ResponseWriter, statusCode int, txt string) {
+	d := ErrorMessage{Error: txt}
+	writeMessage(w, statusCode, d)
+}
+
+// ServeHTTP implements http.Handler
+func (h *deploymentHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	namespace := vars["namespace"]
+	name := vars["name"]
+
+	// check is deployment exist
+	deploymentSpec, err := h.lister.Deployments(namespace).Get(name)
```

Comment:
```suggestion
	deployment, err := h.lister.Deployments(namespace).Get(name)
```
Technically this has both spec and status.

Reply:
done